sdk/errors.go

```diff
@@ -0,0 +1,35 @@
+package sdk
+
+import (
+	"errors"
+
+	mongodbatlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
+)
+
+// GetAPIError gets a strongly typed error from an error.
+func GetAPIError(err error) *mongodbatlasv2.Error {
+	var openapiError mongodbatlasv2.GenericOpenAPIError
+
+	if ok := errors.As(err, &openapiError); ok {
+		errModel := openapiError.Model()
+
+		transformedError, ok := errModel.(mongodbatlasv2.Error)
+		if !ok {
+			return nil
+		}
+		return &transformedError
+	}
+
+	return nil
```

Comment:
[style] I prefer early returns as they avoid deep nesting
```suggestion
	if ok := errors.As(err, &openapiError); !ok {
		return nil	
	}
	errModel := openapiError.Model()
	transformedError, ok := errModel.(mongodbatlasv2.Error)
	if !ok {
		return nil
	}
	return &transformedError
```